internal/server/data/data.go

```diff
@@ -247,18 +277,27 @@ func handleError(err error) error {
 }
 
 func delete[T models.Modelable](db *gorm.DB, id uid.ID) error {
+	if hasOrgID(new(T)) {
+		org := OrgFromContext(db.Statement.Context)
+		db = db.Where("organization_id = ?", org.ID)
+	}
 	return db.Delete(new(T), id).Error
 }
 
 func deleteAll[T models.Modelable](db *gorm.DB, selectors ...SelectorFunc) error {
 	for _, selector := range selectors {
 		db = selector(db)
 	}
+	if hasOrgID(new(T)) {
```

Comment:
Same comment as the other delete function, this looks like it should use the selector also.